crypto/merklekeystore/KeystoreBuilder_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package merklekeystore
+
+import (
+	"runtime"
+	"testing"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/algorand/go-algorand/util/db"
+	"github.com/stretchr/testify/require"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+func TestBuilderSanity(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+
+	numOfKeys := uint64(100)
+	keys, err := KeyStoreBuilder(numOfKeys, crypto.FalconType)
+	a.NoError(err)
+	a.Equal(uint64(len(keys)), numOfKeys)
+
+	s, err := keys[0].GetSigner().SignBytes([]byte{0})
+	v := keys[0].GetSigner().GetVerifyingKey().GetVerifier()
+	err = v.VerifyBytes([]byte{0}, s)
+	a.NoError(err)
+}
+
+func TestBuilderFitsToCPUs(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+	numOfKeys := uint64(runtime.NumCPU() * 2)
+	keys, err := KeyStoreBuilder(numOfKeys, crypto.FalconType)
+	a.NoError(err)
+	a.Equal(uint64(len(keys)), numOfKeys)
+
+}
+
+func TestBuilderOneKey(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+	numOfKeys := uint64(1)
+	keys, err := KeyStoreBuilder(numOfKeys, crypto.FalconType)
+	a.NoError(err)
+	a.Equal(uint64(len(keys)), numOfKeys)
+}
+
+func TestBuilderZeroKeys(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+	numOfKeys := uint64(0)
+	keys, err := KeyStoreBuilder(numOfKeys, crypto.FalconType)
+	a.NoError(err)
+	a.Equal(uint64(len(keys)), numOfKeys)
```

Comment:
`Equal` first argument is the expected value, second is the checked value. The parameter order must be reversed.